File: frontend/src/types/image.ts
export interface ImageInfo {
  filename: string;
  split: string;
  resolution: [number, number] | null;
  has_label: boolean;
  box_count: number;
}

export interface ImageListResponse {
  items: ImageInfo[];
  total: number;
  page: number;
  page_size: number;
}

export interface LabelBox {
  class_id: number;
  class_name: string;
  x_center: number;
  y_center: number;
  width: number;
  height: number;
}

export interface LabelResponse {
  boxes: LabelBox[];
  raw_text: string;
}

File: backend/app/schemas/image.py
from pydantic import BaseModel


class ImageInfo(BaseModel):
    filename: str
    split: str
    resolution: tuple[int, int] | None
    has_label: bool
    box_count: int


class ImageListResponse(BaseModel):
    items: list[ImageInfo]
    total: int
    page: int
    page_size: int


class LabelBox(BaseModel):
    class_id: int
    class_name: str
    x_center: float
    y_center: float
    width: float
    height: float


class LabelResponse(BaseModel):
    boxes: list[LabelBox]
    raw_text: str
